modules/gdscript/gdscript_parser.cpp

```diff
@@ -8211,6 +8215,10 @@ void GDScriptParser::_check_block_types(BlockNode *p_block) {
 							_set_error("The assigned value doesn't have a set type; the variable type can't be inferred.", lv->line);
 							return;
 						}
+						if (assign_type.kind == DataType::BUILTIN && assign_type.builtin_type == Variant::NIL) {
+							_set_error("Invalid assigned value type (\"null\") is not allowed.", lv->line);
```

Comment:
What about this one?

Reply:
@vnen sorry didn't notice, and also changed it now.